src/etc/zed_settings.json

```diff
@@ -0,0 +1,48 @@
+{
+  // The following commented-out VS Code settings are not supported by Zed yet.
+  // "git.detectSubmodulesLimit": 20
```

Comment:
This option was introduced for vscode in https://github.com/rust-lang/rust/pull/120138. There is a fair chance that zed won't suffer from the same issue once it has a git ui anyway (they may not even add a hard limit on the amount of auto detected git submodules at all). I don't think there is any value in keeping this.

Reply:
It could be nicer to still keep one file for RA settings, then just delete this key via bootstrap (serde_json is already a dependency). So if there are future updates that work for both, they only need to happen in one file.

Reply:
In order to do this properly, I think we'll need three files: one shared file for `rust-analyzer` settings and one file each for the non-`rust-analyzer` IDE-specific settings.

Even IDE settings that do the same thing are different between the two IDEs. For example, to turn on format-on-save, VS Code wants `"editor.formatOnSave": true,` while Zed wants `"format_on_save": "on",`.

There are also JSON structure requirements that differ between the IDEs. For example, VS Code wants the `rust-analyzer` settings to use the standard dotted notation at the root of the JSON tree, like:
```json
{
    "rust-analyzer.rustc.source": "discover"
}
```

Zed wants the `rust-analyzer` settings to live under a path through the JSON tree of `"lsp"`, then `"rust-analyzer"`, then `"initialization_options"`, then the setting with the original dots signifying an increase in tree depth:
```json
{
  "lsp": {
    "rust-analyzer": {
      "initialization_options": {
        "rustc": {
          "source": "discover"
        }
      }
    }
  }
}
```

So this means that we need to do JSON parsing of our three files, transformation of the structure, and JSON generation of the resulting `.vscode/settings.json` and `.zed/settings.json` file.

Reply:
You could also just have one rust source file with everything in `json!`, and build it dynamically

```rust
use serde_json::{json, Value};

fn base_settings() -> Value {
    json! {{
         // common settings 
    }}
}

fn vscode_settings() -> Value {
    let vscode_specific = json! {{ anything new}} 

    let mut inner = base_settings();
    inner.as_object_mut().unwrap().extend(vscode_specific);

    json! {{
         "lsp": {
             "rust_analyzer": /* other nesting */ inner,
         }
    }}
}

fn zed_settings() -> Value {
    // ...
}
```

Which would also be kind of nice because it's easy to cover any config file schema or format. E.g. I would appreciate a Helix config at some point, but that uses toml (easy to convert from `serde_json` to `toml`, both are already dependencies of bootstrap).

But 🤷, that may be overkill.